File: study/this_is_coding_interview/dynamic_programing/ant_warrior.py
import sys
N = int(sys.stdin.readline())
storages = list(map(int, sys.stdin.readline().split()))

dp = [0] * (N+1)
dp[1] = storages[0]
for i in range(2, N+1):
  dp[i] = max(dp[i-1], dp[i-2]+storages[i-1])
print(dp)

File: study/this_is_coding_interview/dynamic_programing/ant_warrior.js
const input = require("fs")
  .readFileSync("./input.txt", "utf-8")
  .toString()
  .trim()
  .split("\n");
const N = +input[0];
const storages = input[1].split(" ").map(Number);

const dp = Array.from({ length: N + 1 }, () => 0);
dp[1] = storages[0];
for (let i = 2; i <= N; i++) {
  dp[i] = Math.max(dp[i - 1], dp[i - 2] + storages[i - 1]);
}
console.log(dp);
